x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -1491,6 +1506,27 @@ static void validateApiKeyTypeAndExpiration(
             return;
         }
 
+        if (apiKeyDoc.certificateIdentity != null) {
+            if (credentials.getCertificateIdentity() == null) {
+                listener.onResponse(
+                    AuthenticationResult.terminate("Expected signature for cross cluster API key, but no signature was provided")
```

Comment:
Feel free to argue otherwise, but it feels strange to me that the API Key service knows/assumes that this is a cross cluster API Key just because it has a certificate identity. It feels like we're making this less general purpose than we could.

 
```suggestion
                    AuthenticationResult.terminate(
                        Strings.format("API key (type:[%s], id:[%s]) requires certificate identity [%s], but no certificate was provided", apiKeyDoc.type.value(), credentials.getId(), apiKeyDoc.certificateIdentity));
```